### ISSUE TITLE
Parallelize IPFS Requests for App Data

### ISSUE THREAD

**Issue by nlordell** [cowprotocol/dune-bridge#2]
This iterative async block of code could probably be a batch request.

```rs
    for hash in uninitialized_app_data_hashes.iter() {
        download_referral_from_ipfs_and_store_in_referral_store(db.clone(), *hash).await?;
    }
```

cf `referral_maintenance.rs`

> Original issue https://github.com/gnosis/gp-dune-bridge/issues/59 by @bh2smith

**Comment by josojo** [comment]
Yes, this should be doable. Just a word of warning: When I was first implementing it, I had to fight with rate limitings. If we parallelize it, there is a risk that this will reappear

**Comment by bh2smith** [comment]
To avoid the rate limiting (maybe) You can put a cap on the size of the batches.
